File: benchmark/comparison/results/haiku_ts/09.ts
function processData(lst: number[]): number {
  // Filter negatives
  const filtered = lst.filter(x => x >= 0);
  // Double
  const doubled = filtered.map(x => x * 2);
  // Sum
  return doubled.reduce((sum, x) => sum + x, 0);
}

function run(lst: number[]): void {
  const result = processData(lst);
  console.log(`Total: ${result}`);
}

// Test cases
run([1, 2, 3]);
run([-1, -2]);
run([5, -3, 5]);

File: benchmark/comparison/results/haiku_py/09.py
def process_data(lst):
    # Filter negatives
    filtered = [x for x in lst if x >= 0]
    # Double
    doubled = [x * 2 for x in filtered]
    # Sum
    total = sum(doubled)
    return total

def run(lst):
    result = process_data(lst)
    print(f"Total: {result}")

# Test
run([1, 2, 3])
run([-1, -2])
run([5, -3, 5])
